pkg/migration/controllers/migration.go

```diff
@@ -2735,3 +2738,79 @@ func isObjectHashEqual(dynamicClient dynamic.ResourceInterface, u *unstructured.
 	}
 	return false
 }
+
+// getRelatedCRDListWRTGroupAndComponents takes input of AppRegs
+// and find out all CRDs those need to be collected.
+// The CRD list should contain
+//
+//	a. CRD matching with the required groups.
+//	b. CRDs which are related with the CRDs from above list (#a)
+func getRelatedCRDListWRTGroupAndComponents(client *apiextensionsclient.Clientset, ruleset *inflect.Ruleset, crdList *stork_api.ApplicationRegistrationList, resGroups map[string]string) []*stork_api.ApplicationRegistration {
```

Comment:
nit.. Didn't you want to name the function getRelatedCRDListWRTGroupAndCategories instead of getRelatedCRDListWRTGroupAndComponents?

Reply:
Done